internal/command/release_create.go

```diff
@@ -0,0 +1,200 @@
+package command
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/simplesurance/baur/v3/internal/command/term"
+	"github.com/simplesurance/baur/v3/internal/log"
+	"github.com/simplesurance/baur/v3/pkg/baur"
+	"github.com/simplesurance/baur/v3/pkg/storage"
+
+	"github.com/spf13/cobra"
+)
+
+const releaseCreateExample = `
+baur release create stable		      create a release named 'stable',
+					      including all tasks
+baur release create --include '*.build' v3    create a release named 'v3',
+					      including only tasks called
+					      'build'`
+
+var releaseCreateLongHelp = fmt.Sprintf(`
+Creates a named snapshot of the status of tasks.
+The snapshot includes information about the tasks, their outputs and upload
+destinations.
+
+All tasks included in the release must have a run state of 'Exist'.
+Users can specify a file containing additional metadata, which will be stored
+with the release. It's recommended to keep the amount of metadata small,
+as it is stored in the database.
+
+By default, all tasks are included in the release.
+Pass the --include flag to include only specific tasks.
+See %s for more information about the TARGET syntax.
+
+Exit Codes:
+  %d - Success
+  %d - Error
+  %d - Release with the same name already exist
+  %d - One or more tasks are in pending state
+`,
+	term.Highlight("baur run --help"),
+	exitCodeSuccess,
+	exitCodeError,
+	exitCodeAlreadyExist,
+	exitCodeTaskRunIsPending,
+)
+
+type releaseCreateCmd struct {
+	cobra.Command
+	requireCleanGitWorktree bool
+
+	metadataFile string
+	includes     []string
+}
+
+func init() {
+	releaseCmd.AddCommand(&newReleaseCreateCmd().Command)
+}
+
+func newReleaseCreateCmd() *releaseCreateCmd {
+	cmd := releaseCreateCmd{
+		Command: cobra.Command{
+			Use:     "create NAME",
+			Short:   "create a release",
+			Long:    strings.TrimSpace(releaseCreateLongHelp),
+			Args:    cobra.ExactArgs(1),
+			Example: strings.TrimSpace(releaseCreateExample),
+			ValidArgsFunction: func(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			},
+		},
+	}
+
+	cmd.Flags().BoolVarP(&cmd.requireCleanGitWorktree, flagNameRequireCleanGitWorktree, "c", false,
+		"fail if the git repository contains modified or untracked files")
+	cmd.Flags().StringVarP(&cmd.metadataFile, "metadata", "m", "",
+		"path to a file containing additional data that is stored with the release",
+	)
+
+	const includeFlagName = "include"
+	cmd.Flags().StringArrayVar(&cmd.includes, includeFlagName, nil,
+		"Target to be included in the release, by default all are included.\n"+
+			"It supports the same TARGET syntax as 'baur run',\n"+
+			"the flag can be specified multiple times")
```

Comment:
```suggestion
			"the flag can be specified multiple times.")
```

Reply:
Most flag descriptions aren't whole sentences and do not end in a ".", this one is though.
Not sure if this should then also end up in a "." or be more consistent with the other flag descriptions and not have one at the end

Reply:
Yeah, the other comments were in line with what this flag had.

Reply:
I shortened the description to not being full sentences anymore, new one:
```
      --include stringArray          tasks matching a TARGET string to include in the release,
                                     TARGET has the same syntax as used in 'baur run',
                                     the flag can be specified multiple times
```